prov/gni/src/gnix_eq.c

```diff
@@ -338,9 +333,16 @@ DIRECT_FN STATIC ssize_t gnix_eq_read(struct fid_eq *eq, uint32_t *event,
 
 	eq_priv = container_of(eq, struct gnix_fid_eq, eq_fid);
 
+	if ((blocking && !eq_priv->wait) ||
+	    (blocking && eq_priv->attr.wait_obj == FI_WAIT_SET))
+		return -FI_EINVAL;
+
 	if (_gnix_queue_peek(eq_priv->errors))
 		return -FI_EAVAIL;
 
+	if (eq_priv->wait)
```

Comment:
We could be inside a gnix_eq_read() (non-blocking) even if the EQ has a wait set bound to it.  This check would do the wrong thing in that case.